pkg/api/api.go

```diff
@@ -45,3 +46,17 @@ func NewDefinition() *Definition {
 func (d *Definition) Validate() (bool, error) {
 	return govalidator.ValidateStruct(d)
 }
+
+// UnmarshalJSON api.Definition JSON.Unmarshaller Implementation
+func (d *Definition) UnmarshalJSON(b []byte) error {
+	//Aliasing Definition to avoid recursive call of this method
+	type DefinitionAlias Definition
```

Comment:
Have you considered declaring like this `type definitionAlias Definition` given it is not an exported type?

Reply:
That makes sense. I'll fix it, thanks.